app/scripts/directives/oscRouting.js

```diff
@@ -79,13 +80,20 @@ angular.module("openshiftConsole")
           scope.route.service = _.find(scope.services);
         }
 
-        scope.$watch('route.service', function(newValue, oldValue) {
-          updatePortOptions(scope.route.service);
+        scope.$watch('route.to.service', function(newValue, oldValue) {
+          updatePortOptions(newValue);
           // Don't overwrite the target port when editing an existing route unless the user picked a
           // different service.
           if (newValue !== oldValue || !scope.route.targetPort) {
             scope.route.targetPort = _.get(scope, 'route.portOptions[0].port');
           }
+
+          if (scope.services) {
+            // Update the options for alternate services.
+            scope.alternateServiceOptions = _.reject(scope.services, function(service) {
```

Comment:
what happens if they have already added the service as an alternate and they switch it to be the primary?